### ISSUE TITLE
Fix missing type keyword in schema cases

### ISSUE THREAD

**Issue by JiwoonKim** [taskcluster/material-ui-json-schema-viewer#64]
**Describe the Bug**
- Some schemas seem to miss the `type` keyword
  - ex. `metadata-metaschema.json` which is in our demo and also a Taskcluster schema as well
  - see https://github.com/taskcluster/material-ui-json-schema-viewer/blob/master/schemas/demo/metadata-metaschema.json

This is problematic since the current schema viewer uses the `type` property to render the schemas according to its type. Thus, the above example does not render any properties of the object since it does not explicitly specify its `type`.

**To reproduce**
Currently the `metadata-metaschema.json` example is not included in Storybook's examples. 
Will be able to reproduce this problem once #63 is merged into the master branch. 
In the meantime, refer to the screenshot and schema file comparison example instead (shown below).

**Example**
- the problematic code within `metadata-metaschema.json`:
```js
"recurse": {
    "properties": {
     ...
    }
    // 'type' is not specified 
}
```
- screenshot of this part in `schemaViewer`: does not correctly render the properties even though they are specified since no `type` is given
![image](https://user-images.githubusercontent.com/29671309/73338829-caae6b00-42ba-11ea-9a84-9628d9623c5f.png)

**Comment by djmitche** [comment]
Based on https://json-schema.org/draft/2019-09/json-schema-core.html#rfc.section.7.6.1 I think this is a valid JSON-schema, where the `properties` property of the `recurse` definition only applies to objects.  I'm not sure how best to represent this - it does seem an edge case that will likely occur only in metaschemas and other schema documents that are rarely used as part of user-facing documentation.  Perhaps this is a use for the :warning: icon?

**Comment by JiwoonKim** [comment]
> Based on https://json-schema.org/draft/2019-09/json-schema-core.html#rfc.section.7.6.1 I think this is a valid JSON-schema, where the `properties` property of the `recurse` definition only applies to objects. I'm not sure how best to represent this - it does seem an edge case that will likely occur only in metaschemas and other schema documents that are rarely used as part of user-facing documentation. Perhaps this is a use for the ⚠️ icon?

Thanks for checking that the example above was a valid schema! 👍
Since schemas missing the `type` property is also a valid case, I think we may use a simpler solution:
- If `properties` only applies to objects and `items` or `contains` only applies to array types, how about validating the type of the schema based on whether these properties exist or not? 

At the moment, we're already making use of a custom schema validator which is used to identify the type of the schema based on certain keywords.
https://github.com/taskcluster/material-ui-json-schema-viewer/blob/9c70be1ab7407e3f4ae95ef061da68518e951d7b/src/utils/schemaTree.js#L77-L88
For example, even though combination types (`allOf`, `oneOf`, etc.) schemas do not contain a `type` keyword, we use the `allOf` keyword to add a `type: allOf` property to the schema. Similarly, `$ref` does not specify the `type` either, so we add a `type: $ref` property to the schema as well. (Of course, we create changes to a cloned schema so that the original schema is left immutable :) )

Since we already have this schema validation function, we could also 
- use `properties` to identify object types and add `type: object` to the schemas
- use `items` or `contains` to identify array types and add `type: array` to the schemas

This way, we can simply make use of the custom schema validator we already have while also properly displaying the schemas in question (like meta-schemas) just like other schemas without much issues.

What do you think about this idea?

**Comment by djmitche** [comment]
As I understand it, the difference between
* `{"type": "object", "properties": {"someProp": ..}}`
* `{"properties": {"someProp": ..}}`
is that the first case *requires* that the value is an object while the other does not -- it can be any type, but *if* it is an object, its someProp property must match `..`.  So a value of `13` for example is not valid for the first schema, but valid for the second.

Looking at it a different way, I think the absence of a `type` property in a schema is equivalent to `["null", "boolean", "object", "array", "number", or "string"]` - that is, any type is allowed.

**Comment by JiwoonKim** [comment]
I see, then it would be better to use the ⚠️ icon for such exceptional cases instead. So if a `type` is not specified, we will use the ⚠️ icon in the left row and provide a tooltip that the type is not specified, and direct the user to refer to the source instead. 

This will also apply to other exceptional cases where the type is not specified, such as in the `allOf` example schema (https://github.com/taskcluster/material-ui-json-schema-viewer/blob/master/schemas/combinationTypes/allOf.json) so that it displays:
```
// All of       |
   'string'     |
   // and       |
   ⚠️           |  maxLength: 5

**Comment by djmitche** [comment]
That sounds good.  I think that's probably a good strategy to apply to any remaining "weird" cases at this point.  In use, we'll get some nice GH issues with examples of schemas triggering these cases and room for discussion of how best to represent them, and can tackle them one-by-one.
